homework-submission/todo-app/src/component/week3Components/index.js

```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from "react";
+import Name from "./name";
+import Img from "./img";
+import Region from "./region";
+import Gender from "./gender";
+import "core-js";
+
+function Week3Components() {
+  const [data, setData] = useState("");
+  const [id, setId] = useState(-1);
+
+  useEffect(() => {
+    const fetchData = async url => {
+      await fetch(url)
+        .then(response => response.json())
+        .then(content => setData(content));
+    };
+    try {
+      fetchData("https://uinames.com/api/?amount=10");
+    } catch (err) {
+      console.log(err);
+    }
+
+    return () => fetchData("https://uinames.com/api/?amount=10");
+  }, []);
+
+  document.body.onkeyup = e => {
+    if (e.keyCode === 32 && id <= 8) {
+      setId(id + 1);
+    } else {
+      setId(0);
+    }
+  };
+
+  return (
+    <div className="week3">
+      {id === -1 ? (
+        <h1 className="press-space">Please press space to get new data</h1>
+      ) : (
+        <>
+          <h1>week3-homework</h1>
+          <ul>
+            <Img
+              imgSrc={`https://uinames.com/api/photos/${data[id].gender}/${id +
```

Comment:
If you read the api documentation you can find how to get an image url in the returned json data. A bit neater than generating the url yourself. Please give it a try :)

Reply:
fixed with ext.